packages/contentstack-migration/examples/05-Update-reference-entry-from-mapper.js

```diff
@@ -0,0 +1,156 @@
+const fs = require('fs');
+const chalk = require('chalk');
+const path = require('path');
+module.exports = async ({ migration, stackSDKInstance, managementAPIClient, config }) => {
+  const modules = ['entries', 'assets', 'extensions', 'marketplace_apps'];
+
+  const readAllModulesUids = (filePath) => {
+    let uidMapping = {};
+
+    modules.forEach((module) => {
+      const mappingFilePath = path.join(filePath, 'mapper', module, 'uid-mapping.json');
+      if (fs.existsSync(mappingFilePath)) {
+        const mappedIds = JSON.parse(fs.readFileSync(mappingFilePath, 'utf-8'));
+
+        if (module === 'marketplace_apps') {
+          Object.values(mappedIds).forEach((ids) => Object.assign(uidMapping, ids));
+        } else {
+          Object.assign(uidMapping, mappedIds);
+        }
+      }
+    });
+
+    return uidMapping;
+  };
+
+  const getEntries = async (ct) => {
+    try {
+      let entries = [];
+      let skip = 0;
+      let limit = 100;
+      let count = 0;
+      while (skip <= count) {
+        const res = await stackSDKInstance.contentType(ct).entry().query({ include_count: true, skip, limit }).find();
+        count = res.count;
+        skip += limit;
+        entries = [...entries, ...res.items];
+      }
+      return entries;
+    } catch (err) {
+      console.log(chalk.red(`Cannot Fetch Entries`));
+      throw err;
+    }
+  };
+
+  const getContentTypeSchema = async (ct) => {
+    try {
+      return await stackSDKInstance.contentType(ct).fetch();
+    } catch (err) {
+      console.log(chalk.red(`Error in Fetching the Content-type '${ct}' due to ${err.errorMessage}`));
+    }
+  };
+
+  const findAllEntriesUid = (stringifiedEntry) => {
+    let pattern = /\bblt\w*/g;
+    let matches = stringifiedEntry.match(pattern);
+    return matches;
+  };
+
+  const replaceEntriesWithUpdatedUids = (matches, uidMapping, stringifiedEntry) => {
+    let isUpdated = false;
+    matches.forEach((m) => {
+      if (Object.keys(uidMapping).includes(m)) {
```

Comment:
`Object.keys(uidMapping)` can be out of the loop to avoid unnecessary loop as it's constant and it doesn’t require any dynamic value from the loop

Reply:
Done